diff -rupN cfitsio/CMakeLists.txt cfitsio-new/CMakeLists.txt
--- cfitsio/CMakeLists.txt	2020-04-28 15:58:48.493824400 +0200
+++ cfitsio-new/CMakeLists.txt	2020-04-28 16:00:48.591334500 +0200
@@ -30,6 +30,10 @@
 SET(${PROJECT_NAME}_SOVERSION ${${PROJECT_NAME}_MAJOR_VERSION})
 SET(LIB_NAME cfitsio)
 
+# Define compiler flags for >2GB files
+set(CMAKE_C_FLAGS "-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES")
+set(CMAKE_CXX_FLAGS "-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGE_FILES")
+
 # Microsoft Visual Studio:
 IF(MSVC OR BORLAND)
     # Define
